modules/btcpay/src/Repository/CustomerThreadRepository.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace BTCPay\Repository;
+
+use BTCPay\Factory\CustomerThread;
+
+class CustomerThreadRepository
+{
+	/**
+	 * @throws \PrestaShopException
+	 * @throws \PrestaShopDatabaseException
+	 */
+	public static function fetchOrCreateByByEmailAndOrder(\Shop $shop, \Order $order): \CustomerThread
```

Comment:
The naming here seems not to be what the passed parameters suggest? More like `fetchOrCreateByStoreAndOrder`? but maybe I'm missing some context here

Reply:
Ah ok I can see it below in line 19, it is more related to email and order id

Reply:
We need the customer, the order and the shop to be able to create what we need. It is indeed misleading, but since we can get the email from the order and we need the order anyway, I didn't include it.

Not sure how do it better

Reply:
Yes, makes sense in what it should do but parameters do not show that clearly, in line19 it makes perfect sense though, so maybe leave it as is or go for more generic like "fetchOrCreate()" but ymmv